src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1744,85 +1743,122 @@ internal static void CheckNullableReferenceTypeMismatchOnImplementingMember(Type
                 }
                 else
                 {
-                    ReportMismatchInReturnType<(TypeSymbol implementingType, bool isExplicit)> reportMismatchInReturnType =
-                        (diagnostics, implementedMethod, implementingMethod, topLevel, arg) =>
-                        {
-                            if (arg.isExplicit)
-                            {
-                                // We use ConstructedFrom symbols here and below to not leak methods with Ignored annotations in type arguments
-                                // into diagnostics
-                                diagnostics.Add(topLevel ?
-                                                    ErrorCode.WRN_TopLevelNullabilityMismatchInReturnTypeOnExplicitImplementation :
-                                                    ErrorCode.WRN_NullabilityMismatchInReturnTypeOnExplicitImplementation,
-                                                implementingMethod.Locations[0], new FormattedSymbol(implementedMethod.ConstructedFrom, SymbolDisplayFormat.MinimallyQualifiedFormat));
-                            }
-                            else
+                    static void checkMethodOverride(
+                        TypeSymbol implementingType,
+                        MethodSymbol implementedMethod,
+                        MethodSymbol implementingMethod,
+                        bool isExplicit,
+                        bool checkReturnType,
+                        bool checkParameterTypes,
+                        BindingDiagnosticBag diagnostics)
+                    {
+                        ReportMismatchInReturnType<(TypeSymbol implementingType, bool isExplicit)> reportMismatchInReturnType =
```

Comment:
the best thing for readability here might be if these were simply static local functions at the same level as 'checkMethodOverride'. Then 'checkMethodOverride' would contain just the bits starting at 1802. I think this wouldn't result in more delegate allocations than we have here currently.

It looks like these were kept here to reduce the diff, but it also might be better to just go ahead and move them to the end of the containing method body.

Reply:
I will leave this as-is, to reduce the diffs. But if this was new code, I agree it would make more sense to have several local functions, all at the same nesting level, and all at the bottom of the method.